cfme/web_ui/__init__.py

```diff
@@ -3238,3 +3272,112 @@ def to_emails(self, emails):
 @fill.method((EmailSelectForm, tuple))
 def fill_email_select_form(form, emails):
     form.to_emails = emails
+
+
+class BootstrapSwitch(object):
+    def __init__(self, input_id):
+        """A Bootstrap On/Off switch
+
+        Args:
+            input_id: The HTML ID of the input element associated with the checkbox
+        """
+        self.input_id = input_id
+        self.loc_container = "//input[@id={}]/..".format(quoteattr(self.input_id))
+        self.on_off = "{}/span[contains(@class, 'bootstrap-switch-handle-{}')]".format(
+            self.loc_container, '{}')
+
+    def fill(self, val):
+        """Convenience function"""
+        if val:
+            self.check()
+        else:
+            self.uncheck()
+
+    def check(self):
+        """Checks the bootstrap box"""
+        el = sel.element(self.on_off.format("off"))
+        sel.click(el)
+
+    def uncheck(self):
+        """Unchecks the bootstrap box"""
+        el = sel.element(self.on_off.format("on"))
+        sel.click(el)
+
+    def is_selected(self):
+        if sel.is_displayed("//div[contains(@class, 'bootstrap-switch-on')]{}"
+                .format(self.loc_container)):
+            return True
+        else:
+            return False
+
+
+@fill.method((BootstrapSwitch, bool))
+def fill_bootstrap_switch(bs, val):
+    bs.fill(val)
+
+
+class OldCheckbox(object):
+    def __init__(self, input_id):
+        """An original HTML checkbox element
+
+        Args:
+            input_id: The HTML ID of the input element associated with the checkbox
+        """
+        self.input_id = input_id
+        self.locator = "//input[@id={}]".format(quoteattr(input_id))
+
+    def fill(self, val):
+        """
+        Checks or unchecks
+
+        Args:
+            value: The value the checkbox should represent as a bool (or None to do nothing)
+
+        Returns: Previous state of the checkbox
+        """
+
+        if val is not None:
+            selected = self.is_selected()
+
+            if selected is not val:
+                logger.debug("Setting checkbox {} to {}".format(str(self.locator), str(val)))
+                sel.click(self._el)
+            return selected
+
+    def check(self):
+        """Convenience function"""
+        self.fill(True)
+
+    def uncheck(self):
+        """Convenience function"""
+        self.fill(False)
+
+    def _el(self):
+        return sel.move_to_element(self.locator)
+
+    def is_selected(self):
+        return self._el.is_selected()
```

Comment:
Hmm, `._el` is a function, not a property. I believe you forgot to make it a property. `.is_selected` might also be a property ...